packages/backend/src/ee/services/AiAgentService.ts

```diff
@@ -1450,6 +1460,64 @@ export class AiAgentService {
             return filteredExplore;
         };
 
+        const findFields: FindFieldFn = async ({ fieldSearchQuery }) => {
+            const userAttributes =
+                await this.userAttributesModel.getAttributeValuesForOrgMember({
+                    organizationUuid,
+                    userUuid: user.userUuid,
+                });
+
+            const agentSettings = await this.getAgentSettings(user, prompt);
+
+            const catalogItems = await this.catalogService.searchCatalog({
+                projectUuid,
+                catalogSearch: {
+                    type: CatalogType.Field,
+                    searchQuery: fieldSearchQuery.name,
+                },
+                context: CatalogSearchContext.AI_AGENT,
+                // TODO: make this paginated
+                paginateArgs: {
+                    page: 1,
+                    pageSize: 5,
+                },
+                userAttributes,
+                yamlTags: agentSettings.tags,
+            });
+
+            // TODO: we should not filter here, we should return all the fields
+            const catalogFields = catalogItems.data.filter(
+                (item) => item.type === CatalogType.Field,
+            );
+
+            const explores = await this.projectService.findExplores({
+                user,
+                projectUuid,
+                exploreNames: catalogFields.map((s) => s.tableName),
+            });
+
+            const filteredExplores = Object.values(explores).filter(
+                (e): e is Explore => !isExploreError(e),
+            );
```

Comment:
do we need to check if filteredExplores are empty? 🤔

Reply:
I want to make improvements in a follow-up PR where catalog returns joined explore instead of me manually attaching it here